.github/workflows/ci.yaml

```
@@ -5,17 +5,42 @@ on:
     branches:
       - master
   pull_request:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
 jobs:
   build:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest] # , macos-latest]
+        os: [x86_64-linux]
     steps:
       - uses: actions/checkout@v4
-      - uses: DeterminateSystems/nix-installer-action@main
-      - uses: DeterminateSystems/magic-nix-cache-action@main
-      - name: Install omnix
-        run: nix profile install nixpkgs#omnix
       - name: Build all flake outputs
         run: om ci
+      - name: Prepare static site
+        if: github.ref == 'refs/heads/master'
+        run: nix build .#site-github -o ./_site
```

When using custom domain, we gotta change this to:

```suggestion
        run: nix build .#site -o ./_site
```